src/s1reader/s1_orbit.py

```diff
@@ -337,7 +397,271 @@ def get_orbit_file_from_list(zip_path: str, orbit_file_list: list) -> str:
+
+    # get file name and extract state vector start/end time strings
+    t_orbit_start, t_orbit_stop = os.path.basename(orbit_file).split('_')[-2:]
+
+    # strip 'V' at start of start time string
+    t_orbit_start = datetime.datetime.strptime(t_orbit_start[1:], FMT)
+
+    # string '.EOF' from end of end time string
+    t_orbit_stop = datetime.datetime.strptime(t_orbit_stop[:-4], FMT)
+
+    # check if:
+    # 1. swath start and stop time > orbit file start time
+    # 2. swath start and stop time < orbit file stop time
+    return all(t_orbit_start < t < t_orbit_stop for t in t_start_stop_frame)
+
+
+def get_resorb_pair_from_list(zip_path: str, orbit_file_list: list,
+                               concatenate_resorb: bool=False) -> list | str | None:
+    '''
+    Find if there are two RESORB files that cover [start - margin_start_time, end]
+    If found, try to concatenate them when `concatenate_resorb` is `True`
+
+    Code based on `get_orbit_file_from_list()`
+
+    Parameters
+    ----------
+    zip_path : string
+        Path to the Sentinel-1 SAFE zip file. Base names required to adhere to the
+        format described here:
+        https://sentinel.esa.int/web/sentinel/user-guides/sentinel-1-sar/naming-conventions
+    orbit_file_list : list
+        List of the orbit files that exist in the system.
+
+    Returns
+    -------
+    orbit_file : str
+        Path to the orbit file, or `None` if no orbit file was found.
+    '''
+    # check the existence of input file path and directory
+    if not os.path.exists(zip_path):
+        raise FileNotFoundError(f"{zip_path} does not exist")
+
+    # extract platform id, start and end times from swath file name
+    mission_id, t_swath_start_stop = get_file_name_tokens(zip_path)
+
+    # choose the RESORB file only
+    resorb_file_list = [orbit_file for orbit_file in orbit_file_list
+                        if '_RESORB_' in os.path.basename(orbit_file)]
+    pad_1min = datetime.timedelta(seconds=PADDING_SHORT)
+    resorb_filename_earlier = None
+    resorb_filename_later = None
+    for resorb_file in resorb_file_list:
+        if mission_id not in os.path.basename(resorb_file):
+            continue
+
+        # 1. Try to find the orbit file that covers
+        # sensing time - margin_start_time
+        t_swath_start_stop_safe = [t_swath_start_stop[0] - pad_1min,
+                                   t_swath_start_stop[1] + pad_1min]
+        if (_covers_timeframe(resorb_file, t_swath_start_stop_safe) and
+                resorb_filename_later is None):
+            print('Found RESOEB file covering the S1 SAFE frame.')
+            resorb_filename_later = resorb_file
+            continue
+
+        # 2. Try to find the orbit file that covers the sensing start-stop
+        # with small padding (like 60 sec.)
+        t_swath_start_stop_anx = [t_swath_start_stop[0] - margin_start_time,
+                                  t_swath_start_stop[0] - margin_start_time + 2*pad_1min]
```

Comment:
I expected to search for time range of "SLC_sensing_start - T_ORB" to "SLC_sensing_start - pad_1min". 
But instead we are searching an orbit to cover "SLC_sensing_start - T_ORB" to "SLC_sensing_start - T_ORB + 2 pad_1min". Again was this the intention really?